srtcore/core.cpp

```diff
@@ -9782,15 +9779,19 @@ int srt::CUDT::handleSocketPacketReception(const vector<CUnit*>& incoming, bool&
             {
                 LOGC(qrlog.Warn, log << CONID() << "No room to store incoming packet seqno " << rpkt.m_iSeqNo
                         << ", insert offset " << offset << ". "
-                        << m_pRcvBuffer->strFullnessState(qrlog.Warn.CheckEnabled(), m_iRcvLastAck, steady_clock::now())
+                        << m_pRcvBuffer->strFullnessState(qrlog.Warn.CheckEnabled(), steady_clock::now())
                     );
 
                 return -1;
             }
         }
-
-        int buffer_add_result = m_pRcvBuffer->insert(u);
-        if (buffer_add_result < 0)
+        else if (buffer_add_result == -2)
+        {
+            // The insert() result is -2 if the packet is behind the start of recv buffer.
+            IF_HEAVY_LOGGING(exc_type = "BELATED");
+            adding_successful = false;
+        }
```

Comment:
This case is interesting. Belated packets are sorted out at the start of the function, including counting the number of belated packets.
Now there are two places they are sorted out but only one updating the stats. It might be a place for mistakes in future maintenance. Please consider combining both somehow.